src/mpid/common/bc/mpidu_bc.c

```diff
@@ -142,7 +144,7 @@ int MPIDU_bc_table_create(int rank, int size, int *nodemap, void *bc, int bc_len
         *ret_bc_len = recv_bc_len;
     }
 
-    mpi_errno = MPIDU_Init_shm_alloc(recv_bc_len * size, (void **) &segment);
+    mpi_errno = MPIDU_Init_shm_alloc(recv_bc_len * (size + local_size), (void **) &segment);
```

Comment:
Please add a comment on the reason -- that we may allgather a local bc table into a global bc table.

Reply:
done